### PR TITLE
feat(nomination): Add opt-out postconditions

### PR DESCRIPTION
**Pull request by savudani8**

### CODE REVIEW

File: interbtc-spec/docs/source/spec/nomination.rst [hunk 1]
```diff
@@ -281,6 +281,11 @@ Specification
 *Postconditions*
 
 * The Vault MUST be removed from the ``Vaults`` mapping.
+* The Vault MUST remain above the secure collateralization threshold.
+* ``get_total_nominated_collateral(vault_id)`` must return zero.
+* For all nominators, ``get_nominator_collateral(vault_id, user_id)`` must return zero.
+* Staking pallet ``nonce`` must be incremented by one.
+* ``compute_reward_at_index(nonce - 1, INTERBTC, vault_id, user_id)`` in the Staking pallet must be equal to the user's nomination just before the vault opted out.
```
Comment (gregdhill) [inline interbtc-spec/docs/source/spec/nomination.rst:288]: Use camelCase for all code please.

File: interbtc-spec/docs/source/spec/nomination.rst [hunk 2]
```diff
@@ -281,6 +281,11 @@ Specification
 *Postconditions*
 
 * The Vault MUST be removed from the ``Vaults`` mapping.
+* The Vault MUST remain above the secure collateralization threshold.
+* ``get_total_nominated_collateral(vault_id)`` must return zero.
+* For all nominators, ``get_nominator_collateral(vault_id, user_id)`` must return zero.
```
Comment (gregdhill) [inline interbtc-spec/docs/source/spec/nomination.rst:286]: Are you able to `:ref:` these?

**Review by gregdhill — CHANGES_REQUESTED** [review]

File: interbtc-spec/docs/source/spec/nomination.rst (hunk 2 above)
Comment (savudani8) [reply interbtc-spec/docs/source/spec/nomination.rst:286]: Being getters, these were not added to the spec

Comment (gregdhill) [reply interbtc-spec/docs/source/spec/nomination.rst]: I think `computeRewardAtIndex` should be ref-able right?

**Review by gregdhill — APPROVED** [review]

**Pull request merged**